tekton/ci/jobs/tekton-catlin-lint.yaml

```diff
@@ -34,11 +34,32 @@ spec:
       image: gcr.io/tekton-releases/dogfooding/catlin:latest
       workingDir: $(resources.inputs.source.path)
       script: |
+        set +e
         [[ ! -s $(workspaces.store-changed-files.path)/changed-files.txt ]] && {
           echo "No file change detected in task directory"
+          echo -n "" > $(workspaces.store-changed-files.path)/catlin.txt
           exit 0
         }
-        catlin validate $(cat $(workspaces.store-changed-files.path)/changed-files.txt)
+
+        # creating a file which will contain the final formatted output
+        # which needs to be added as a comment(if any)
+        echo '**Catlin Output**' >> $(workspaces.store-changed-files.path)/catlin.txt
+        echo '```' >> $(workspaces.store-changed-files.path)/catlin.txt
+
+        # performing catlin validate
+        catlin validate $(cat $(workspaces.store-changed-files.path)/changed-files.txt) | tee -a $(workspaces.store-changed-files.path)/catlin.txt
+        echo '```' >> $(workspaces.store-changed-files.path)/catlin.txt
+
+        # checking if there is any ERROR or WARN produces by catlin
+        isWarning=$(cat  $(workspaces.store-changed-files.path)/catlin.txt | grep -c "WARN")
+        isError=$(cat  $(workspaces.store-changed-files.path)/catlin.txt | grep -c "ERROR")
+
+        # if there is any ERROR or WARN then post the comment else skip
+        [[ $isWarning -eq 0 ]] && [[ $isError -eq 1 ]] && \
+        echo -n "" > $(workspaces.store-changed-files.path)/catlin.txt
```

Comment:
I don't really understand the logic here, perhaps it should be `$isError -eq 0`? 

BTW, it looks like the `post-comment` will be executed regardless, only with an empty comment?
Hopefully the github API is smart enough to not post an empty comment?

A nice solution would be to have a result and use `when` to skip the comment if not needed. Unfortunately `when` expressions are not available in `finally` yet (https://github.com/tektoncd/community/pull/240).

Reply:
> I don't really understand the logic here, perhaps it should be `$isError -eq 0`?
> 
oh yes :sweat_smile:, it should be `$isError -eq 0`. Will fix this :sweat_smile: 
> BTW, it looks like the `post-comment` will be executed regardless, only with an empty comment?
> Hopefully the github API is smart enough to not post an empty comment?
> 
yeah when there is an empty comment then no comment would be posted on PR

> A nice solution would be to have a result and use `when` to skip the comment if not needed. Unfortunately `when` expressions are not available in `finally` yet ([tektoncd/community#240](https://github.com/tektoncd/community/pull/240)).

Reply:
btw the following script will produce the output as 

----
**Catlin Output**
```
FILE: task/black/0.1/black.yaml
WARN : Step "format-python-code" uses image "$(params.image)" that contains variables; skipping validation
FILE: task/yaml-lint/0.1/yaml-lint.yaml
```